From fbcde67f433db1b36e47e0cfcc735f8c661be294 Mon Sep 17 00:00:00 2001
From: Changqing Li <changqing.li@windriver.com>
Date: Mon, 23 Mar 2026 05:48:14 +0000
Subject: [PATCH] fix kleidiai download failure

Seems https://gitlab.arm.com only allow TLS v1.3 verification, set
TLS_VERSION to 1.3 to fix dowload failure:
error: downloading 'https://gitlab.arm.com/kleidi/kleidiai/-/archive/847ebd19d0192528659b0a0fa2c6057eed674c6a/kleidiai-847ebd19d0192528659b0a0fa2c6057eed674c6a.zip' failed
          status_code: 35
          status_string: "SSL connect error.  If this is due to https certificate verification failure, one may set environment variable CMAKE_TLS_VERIFY=0 to suppress it.

Upstream-Status: Pending

Signed-off-by: Changqing Li <changqing.li@windriver.com>

---
 tensorflow/lite/cmake/DownloadKleidiAI.cmake | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tensorflow/lite/cmake/DownloadKleidiAI.cmake b/tensorflow/lite/cmake/DownloadKleidiAI.cmake
index 58b94590..9d519266 100644
--- a/tensorflow/lite/cmake/DownloadKleidiAI.cmake
+++ b/tensorflow/lite/cmake/DownloadKleidiAI.cmake
@@ -21,6 +21,7 @@ INCLUDE(ExternalProject)
 ExternalProject_Add(kleidiai
   URL https://gitlab.arm.com/kleidi/kleidiai/-/archive/847ebd19d0192528659b0a0fa2c6057eed674c6a/kleidiai-847ebd19d0192528659b0a0fa2c6057eed674c6a.zip
   URL_HASH SHA256=3564707756473a165ddb3cf670187a4eeb2189c71c5ef01e865567cab071daac
+  TLS_VERSION 1.3
   SOURCE_DIR "${CMAKE_BINARY_DIR}/kleidiai-source"
   BINARY_DIR "${CMAKE_BINARY_DIR}/kleidiai"
   CONFIGURE_COMMAND ""
-- 
2.49.1

